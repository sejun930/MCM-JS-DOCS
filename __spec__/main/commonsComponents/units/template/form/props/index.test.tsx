import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";

import _PropsForm from "src/main/commonsComponents/units/template/form/props";

export default describe("Props Template Page", () => {
  // 스냅샷
  test("Snapshot", () => {
    const { container } = render(
      <RecoilRoot>
        <_PropsForm />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });

  // h2 태그 검증
  test("Check h2 tag", () => {
    const { container } = render(
      <RecoilRoot>
        <_PropsForm />
      </RecoilRoot>
    );

    const h2Ele = container.querySelector(".subTitle");
    expect(h2Ele).toBeInTheDocument();

    // h2 태그와 텍스트 검증
    if (h2Ele) {
      expect(h2Ele.textContent).toEqual("📤 Props List");
    }

    const remarksEle = container.querySelector(".subTitle-remarks");
    expect(remarksEle).toBeInTheDocument();

    // remarks 텍스트 검증
    if (remarksEle) {
      expect(remarksEle.textContent).toEqual(
        "Props들을 이용해 원하는 모듈을 조립해보세요."
      );
    }
  });

  // props list 페이지가 존재하는지 검증
  test("Check props list wrapper", () => {
    const { container } = render(
      <RecoilRoot>
        <_PropsForm />
      </RecoilRoot>
    );

    const listWrapper = container.querySelector(".props-empty-list-wrapper");
    expect(listWrapper).toBeInTheDocument();
  });
});
