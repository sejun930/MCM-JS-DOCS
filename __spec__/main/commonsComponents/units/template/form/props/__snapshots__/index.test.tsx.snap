// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Props Template Page Snapshot 1`] = `
<div>
  <div
    class="sc-dmqHEX iHpFyK"
  >
    <div
      class="subTitle-Template props-subTitle css-cw3rkw"
    >
      <div
        class="subTitle-title-wrapper css-1knbux5"
      >
        <h2
          class="mcm-title-unit subTitle"
        >
          📤 Props List
        </h2>
      </div>
      <div
        class="subTitle-line css-18klhw4"
      />
      <p
        class="mcm-p-unit subTitle-remarks css-z401r8"
      >
        Props들을 이용해 원하는 모듈을 조립해보세요.
      </p>
    </div>
    <div
      class="props-empty-list-wrapper"
    />
  </div>
</div>
`;
